src/js/covidMap.js

```diff
@@ -585,6 +558,37 @@ fluid.defaults("fluid.covidMap.map", {
     }
 });
 
+// When the location button on the mobile view is activated.
+fluid.covidMap.map.locationButtonOnMobileActivate = function (map, citiesList) {
+    if (citiesList.expandButton.model.expanded) {
+        citiesList.expandButton.applier.change("expanded", false);
+    } else {
+        citiesList.expandButton.applier.change("expanded", true);
+        map.applier.change("query", "");
+        map.applier.change("visiblePanelOnMobileFlags", {
+            "citiesList": true,
+            "resultsPage": false,
+            "filterPanel": false,
+            "hospitalPanel": false
+        });
+    }
+};
+
+// When the filter button on the mobile view is activated.
+fluid.covidMap.map.filterButtonOnMobileActivate = function (map, filterPanel) {
+    if (filterPanel.expandButton.model.expanded) {
```

Comment:
We can absorb the "toggle" part of this interaction in a clearer design using `fluid.transforms.toggle` e.g.
````
        filterButtonOnMobile: {
            type: "fluid.button",
            container: "{map}.dom.filterButtonOnMobile",
            options: {
                modelRelay: {
                    source: "activate",
                    target: "{filterPanel}.expandButton.model.expanded",
                    transform: "fluid.transforms.toggle"
                }
````
Please can you then absorb the remaining part of the interaction with a 

````
    modeListeners: {
        showFilterPanel: {
            path: "{filterPanel}.expandButton.model.expanded",
            listener: "fluid.covidMap.map.filterPanelOnMobileShow",
            args: ["{change}.value", "{map}"]
        }
fluid.covidMap.map.filterButtonOnMobileShow = function (show, map) {
    if (show) {
        map.applier.change("visiblePanelOnMobileFlags", {
            "citiesList": false,
            "resultsPage": false,
            "filterPanel": true,
            "hospitalPanel": false
        });
    }
}
````

And then please leave a comment by the implementation saying
````
// This listener can be improved once "gated listeners" are implemented for https://issues.fluidproject.org/browse/FLUID-6393
````

Thanks!